http/src/main/java/com/redhat/lightblue/client/http/LightblueHttpClient.java

```diff
@@ -18,17 +14,17 @@
 import com.redhat.lightblue.client.LightblueClient;
 import com.redhat.lightblue.client.LightblueClientConfiguration;
 import com.redhat.lightblue.client.PropertiesLightblueClientConfiguration;
-import com.redhat.lightblue.client.http.auth.HttpClientCertAuth;
-import com.redhat.lightblue.client.http.auth.HttpClientNoAuth;
-import com.redhat.lightblue.client.http.request.LightblueHttpDataRequest;
-import com.redhat.lightblue.client.http.request.LightblueHttpMetadataRequest;
+import com.redhat.lightblue.client.http.transport.HttpClient;
+import com.redhat.lightblue.client.http.transport.JavaNetHttpClient;
 import com.redhat.lightblue.client.request.AbstractLightblueDataRequest;
-import com.redhat.lightblue.client.request.AbstractLightblueMetadataRequest;
+import com.redhat.lightblue.client.request.LightblueRequest;
+import com.redhat.lightblue.client.response.DefaultLightblueResponse;
 import com.redhat.lightblue.client.response.LightblueResponse;
 import com.redhat.lightblue.client.response.LightblueResponseParseException;
 import com.redhat.lightblue.client.util.JSON;
 
-public class LightblueHttpClient implements LightblueClient {
+public class LightblueHttpClient implements LightblueClient, Closeable {
```

Comment:
The internal, Apache http client specific were gutted for a generic `HttpClient` interface that just has an `executeRequest` method. This class also by default uses a java.net implementation of the new `HttpClient` interface which is, in my tests, around 70% faster for SSL requests.